pkg/internal/tests/cross-tests/configure.go

```diff
@@ -0,0 +1,155 @@
+// Copyright 2016-2024, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+
+package crosstests
+
+import (
+	"context"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/resource"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/util/contract"
+	"github.com/stretchr/testify/require"
+	"github.com/zclconf/go-cty/cty"
+	"gopkg.in/yaml.v3"
+
+	"github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tests/pulcheck"
+	"github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tests/tfcheck"
+	"github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfbridge/info"
+	shimv2 "github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfshim/sdk-v2"
+)
+
+func MakeConfigure(
+	provider map[string]*schema.Schema, tfConfig cty.Value, puConfig resource.PropertyMap,
+	options ...ConfigureOption,
+) func(*testing.T) {
+	return func(t *testing.T) { Configure(t, provider, tfConfig, puConfig, options...) }
+}
+
+// Configure validates that a Terraform provider witnesses the same input when:
+// - invoked directly with HCL on tfConfig
+// - bridged and invoked via Pulumi YAML on puConfig
+//
+// Create only applies to resources defined with github.com/hashicorp/terraform-plugin-sdk/v2. For cross-tests
+// on Plugin Framework based resources, see
+// github.com/pulumi/pulumi-terraform-bridge/pkg/pf/tests/internal/cross-tests.
+func Configure(
+	t T, provider map[string]*schema.Schema, tfConfig cty.Value, puConfig resource.PropertyMap,
+	options ...ConfigureOption,
+) {
+
+	var opts configureOpts
+	for _, f := range options {
+		f(&opts)
+	}
+
+	if shouldInferredPulumiValue(puConfig) {
+		puConfig = inferPulumiValue(t,
+			shimv2.NewSchemaMap(provider),
+			opts.resourceInfo.GetFields(),
+			tfConfig,
+		)
+	}
+
+	tfwd := t.TempDir()
+	const configureResult = "some-value"
```

Comment:
It's hard coded because it's an opaque value to the TF framework, and so we don't need to test different values.

We only need to test that it was propagated correctly.